[RFC] fix, extend & standardize configdata descriptions

mschilli87 opened the pull request:
This PR contains several changes of the configdata descriptions:

 * add units where appropriate (fixes #3171)
 * apply trivial fixes:
   * fixed typos
   * add missing periods (.) add the end of descriptions
   * fix capitalization of URL where appropriate
   * break long lines (max. 79 chars as per PEP8)
 * replace abbreviations by full expressions:
   * ms -> milliseconds
   * px -> pixels
   * regexes -> regular expressions
   * chars -> characters
 * standardize description format:
   * encapsulate units in parenthesis
   * move units right behind the corresponding noun
   * add articles where appropriate
   * use 'whether' for bools (& boolAsks)
   * use 'when' for selections
   * drop phrases referring to the settings themselves ('set', 'control', ...)

Originally I just wanted to work on #3171 but I found some obvious mistakes and quite some inconsistencies across the descriptions and ended up standardizing them (using the most commonly used style as the default).

I marked it work-in-progress as I didn't double-check myself if I missed anything yet, but feedback would be appreciated.

Also, if it helps review, I could split this PR into several commits.

Obviously, this is super low priority.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qutebrowser/qutebrowser/3181)
<!-- Reviewable:end -->

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -140,7 +140,7 @@ qt.force_software_rendering:
   backend: QtWebEngine
   desc: >-
-    Force software rendering for QtWebEngine.
+    Whether to force software rendering for QtWebEngine.
```
I'd like to keep the descriptions in [imperative mood](https://en.wikipedia.org/wiki/Imperative_mood) if possible, without adding "Whether to". They just add unnecessary fluff here IMHO, which makes things slightly harder to read, and takes away space in the completion.

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -291,15 +292,15 @@ content.headers.custom:
       encoding: ascii
     none_ok: true
-  desc: Set custom headers for qutebrowser HTTP requests.
+  desc: The custom headers for qutebrowser HTTP requests.
```
This sounds weird to me. Maybe "Add custom headers", changing this to imperative mood?

mschilli87 replied:
@The-Compiler: I thought about this again and in the latest version we ended up using imperative mood for boolean settings.
That's not really the case here and I think now that I changed this to "Add custom headers" it kind of implies I can choose whether or not to add a custom header here. This is not true as really I can choose the (potentially empty) content of that header.
As we dropped the articles for most of those cases, how about changing this to "Custom headers to add for qutebrowser HTTP requests." or even simply "Custom headers for qutebrowser HTTP requests." instead?

The-Compiler replied:
"Custom headers for qutebrowser HTTP requests." sounds good to me!

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -389,7 +390,7 @@ content.host_blocking.lists:
     none_ok: true
   desc: |
-    List of URLs of lists which contain hosts to block.
+    The list of URLs of lists which contain hosts to block.
```
This sounds odd. I'd rather keep the old form.

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -406,7 +407,7 @@ content.host_blocking.whitelist:
     none_ok: true
   desc: >-
-    List of domains that should always be loaded, despite being ad-blocked.
+    The list of domains that should always be loaded, despite being ad-blocked.
```
This sounds odd. I'd rather just remove "List of".

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -509,23 +512,23 @@ content.media_capture:
     none_ok: true
   desc: >-
-    Location of a netrc-file for HTTP authentication.
+    The location of a netrc-file for HTTP authentication.
```
"A netrc-file to use for HTTP authentication."?

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -571,31 +574,31 @@ content.proxy_dns_requests:
     none_ok: True
   default: []
-  desc: A list of user stylesheet filenames to use.
+  desc: The list of user stylesheet filenames to use.
```
This sounds odd, I prefer the old form (or maybe just "List of ...")

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -643,22 +647,22 @@ completion.show:
     name: Int
     minval: 0
-  desc: Padding of scrollbar handle in the completion window (in px).
+  desc: The padding (in pixels) of scrollbar handle in the completion window.
```
While you're at it: "of scrollbar handle" -> "of the scrollbar handle"

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -790,19 +795,18 @@ hints.auto_follow:
-  desc: A timeout (in milliseconds) to ignore normal-mode key bindings after a
-    successful auto-follow.
+  desc: The timeout (in milliseconds) to ignore normal-mode key bindings after
+    a successful auto-follow.
```
The duration (...) to ignore"?

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1093,7 +1098,7 @@ spellcheck.languages:
   default: []
   desc: >-
-    Spell checking languages.
+    The spell checking languages.
```
"Languages to use for spell checking."?

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1197,7 +1202,7 @@ tabs.padding:
     right: 5
   type: Padding
-  desc: Padding around text for tabs
+  desc: The padding (in pixels?) around text for tabs
```
yes, in pixels :wink:

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -223,21 +224,21 @@ content.cookies.accept:
   default: iso-8859-1
   desc: >-
-    Default encoding to use for websites.
+    The default encoding to use for websites.
```
Same here - I don't think the "The " prefix adds anything substantial. Some things definitely sound worse than before, IMHO.

mschilli87 commented:
@The-Compiler: Thx for the review, I'll adress your suggestions and re-base fixing the merge conflict.
Is this OK as a single commit or would you rather have it split?

The-Compiler commented:
Both works for me - whatever you prefer.

mschilli87 commented:
I rebased on latest master and applied the requested changes:

 * dropped 'Whether' and 'The' prefixes
 * switched to imperative mood where appropriate
 * applied some re-phrasing suggested by @The-Compiler
 * removed leftover '?'

I don't really care about the exact phrasing, I'd just like to have it standardized.
I originally added 'The' and 'Whether' as I saw it like that in many places and I preferred to add information over discarding it.
This way it was easy to drop those prefixes using regexes as the corresponding lines were already standardized.

Please note that this is still WIP and I will rebase/squash and most likely split again to make sure I didn;t miss anything.

I'd just like to get another round of review if now the changes look good.

jgkamat commented on qutebrowser/config/configdata.yml:
```diff
@@ -432,7 +433,7 @@ content.javascript.can_access_clipboard:
   type: Bool
   desc: >-
-    Whether JavaScript can read from or write to the clipboard.
+    Enable JavaScript to read from or write to the clipboard.
```
The "Enable javascript to <x>" phrase feels a little weird to me, not sure why.

I think that "Allow javascript to <x>" **might** be better, but it would sacrifice the consistency of naming for boolean values. I'm open to both, just wanted to bring it up.

jgkamat commented on qutebrowser/config/configdata.yml:
```diff
@@ -493,17 +494,17 @@ content.javascript.prompt:
   default: false
   type: Bool
-  desc: Whether locally loaded documents are allowed to access remote urls.
+  desc: Allow locally loaded documents to access remote URLs.
```
If we decide to enforce 'Enable' at the front of every bool, this should be changed. I like this more though, fwiw.

mschilli87 replied:
I think we don't need 'Enable' for every bool as long as it's imperative mood.
@The-Compiler: Do you agree?

The-Compiler replied:
Yeah, and for cases like that (or the JS permissions), I think "Allow ..." is better.

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -239,7 +240,7 @@ content.default_encoding:
     Default encoding to use for websites.
 
-    The encoding must be a string describing an encoding such as _utf-8_,
+    Encoding must be a string describing an encoding such as _utf-8_,
```
This should probably be "The encoding", as it is the long/extended description.

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -391,7 +392,7 @@ content.host_blocking.lists:
     List of URLs of lists which contain hosts to block.
 
-    The file can be in one of the following formats:
+    File can be in one of the following formats:
```
Same as above.

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -416,12 +417,12 @@ content.host_blocking.whitelist:
   default: true
   type: Bool
-  desc: Whether images are automatically loaded in web pages.
+  desc: Load images are automatically in web pages.
```
`s/are //`

jgkamat commented on qutebrowser/config/configdata.yml:
```diff
@@ -869,7 +869,7 @@ hints.next_regexes:
       name: Regex
       flags: IGNORECASE
-  desc: A comma-separated list of regexes to use for 'next' links.
+  desc: A comma-separated list of regular expressions to use for 'next' links.
```
Not sure about this, but dropping the 'A' might be a little more consistent.

"Comma-seperated list of..."

I don't mind if you think it's wrong though...

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -441,17 +442,17 @@ content.javascript.can_close_tabs:
   default: false
   type: Bool
-  desc: Whether JavaScript can open new tabs without user interaction.
+  desc: Enable JavaScript to open new tabs without user interaction.
```
Maybe those should be "Allow JavaScript to ..." instead of "Enable JavaScript to ..."?

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -517,15 +518,15 @@ content.netrc_file:
   type: BoolAsk
   backend: QtWebKit
-  desc: Allow websites to show notifications.
+  desc: Wheter to allow websites to show notifications.
```
Probably missed this one because of the typo? Should probably be the old version.

jgkamat commented on qutebrowser/config/configdata.yml:
```diff
@@ -2324,12 +2329,12 @@ bindings.default:
 
-    The main purpose of this setting is that you can set it to an empty dictionary if you
-    want to load no default keybindings at all.
+    Main purpose of this setting is that you can set it to an empty
```
I think that dropping the `The` in this case and the case below sounds a bit weird as well. Maybe we can make an exeption for longer blocks like this one, since they feel more like a dialouge.

mschilli87 replied:
I agree. This was unintended and I tried to undo those replacements (there where a few more like 'The following ... are supported'). I must have missed this one.

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1135,7 +1136,7 @@ tabs.close_mouse_button:
       - middle: "Close tabs on middle-click."
       - none: "Don't close tabs using the mouse."
-  desc: On which mouse button to close tabs.
+  desc: Mouse button on which to close tabs.
```
on -> with maybe?

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1154,9 +1155,9 @@ tabs.favicons.scale:
+    Scaling factor for favicons in the tab bar.
 
-    The tab size is unchanged, so big favicons also require extra `tabs.padding`.
+    Tab size is unchanged, so big favicons also require extra `tabs.padding`.
```
The tab size

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1262,20 +1263,20 @@ tabs.title.format:
+    * `{perc}`: Percentage as a string like `[10%]`.
+    * `{perc_raw}`: Raw percentage, e.g. `10`
+    * `{title}`: Title of the current web page
+    * `{title_sep}`: String ` - ` if a title is set, empty otherwise.
```
The string

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1293,8 +1294,8 @@ tabs.title.format_pinned:
     none_ok: true
-  desc: The format to use for the tab title for pinned tabs. The same placeholders
-    like for `tabs.title.format` are defined.
+  desc: Format to use for the tab title for pinned tabs. Same placeholders like
```
The same

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1412,26 +1414,26 @@ window.title_format:
-    * `{scroll_pos}`: The page scroll position.
-    * `{host}`: The host of the current web page.
+    Format to use for the window title.
+    Following placeholders are defined:
```
The following

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1412,26 +1414,26 @@ window.title_format:
+    * `{perc}`: Percentage as a string like `[10%]`.
+    * `{perc_raw}`: Raw percentage, e.g. `10`
+    * `{title}`: Title of the current web page
+    * `{title_sep}`: String ` - ` if a title is set, empty otherwise.
```
The string

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -2029,7 +2034,7 @@ fonts.web.size.minimum_logical:
     minval: 0
     maxval: maxint
-  desc: The minimum logical font size that is applied when zooming out.
+  desc: Minimum logical font size (in points) that is applied when zooming out.
```
Looking at [the Qt docs](http://doc.qt.io/qt-5/qwebenginesettings.html#setFontSize), I think those are all in pixels?

mschilli87 replied:
good catch, I remeber looking it uo but must have been asleep.

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -2324,12 +2329,12 @@ bindings.default:
 
-    The main purpose of this setting is that you can set it to an empty dictionary if you
-    want to load no default keybindings at all.
+    Main purpose of this setting is that you can set it to an empty
```
The main

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -2384,10 +2389,10 @@ bindings.commands:
     full list of available commands.
 
-    The following modes are available:
+    Following modes are available:
```
The following

The-Compiler commented:
@jgkamat and I reviewed this at roughly the same time, so I didn't read his comments yet when I made mine - looks like we mostly agree though 😆

mschilli87 commented:
I adressed your comments and re-based on latest master.

I'll have another look to make sure I (or rather we) didn't miss anything else (probably tomorrow) and then split the changes into more logical chunks (typos fixed, units added, specific standardizations) to better document the consensus from this discussion in the commit log.

When I'm done I'll update the status from WIP to RDY.

The-Compiler commented:
Thanks! I'll take another look as well, and maybe @jgkamat too :wink:

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1542,7 +1544,7 @@ colors.completion.scrollbar.fg:
   default: '#333333'
   type: QssColor
-  desc: Color of the scrollbar in completion view
+  desc: Color of the scrollbar in completion view.
```
"in *the* completion view." probably?

mschilli87 commented:
I started splitting this into logical chunks, checking each of them again and actually found (a lot) more along the way. I'll finish my review (hopefully) tomorrow and push the (presumably) final version for review when I'm done.
@jgkamat: Maybe you want to wait for my push to avoid suggesting changes I already implemented locally. :wink:

The-Compiler commented:
Thanks for the great work! 👍 

What do you think about adding you to the `.github/CODEOWNERS` file for `qutebrowser/config/configdata.yml`? Then you'll get notified for any pull requests which touch it, which would probably help to maintain consistency :wink:

If you can see yourself doing this, can you please add yourself to the file in this PR?

mschilli87 commented on qutebrowser/config/configdata.yml:
```diff
@@ -611,7 +611,7 @@ completion.cmd_history_max_items:
     minval: -1
   desc: >-
-    How many commands to save in the command history.
+    Number of commands to save in the command history.
```
here we prefer a noun over 'How ...'

mschilli87 commented on qutebrowser/config/configdata.yml:
```diff
@@ -1182,7 +1183,7 @@ tabs.last_close:
       - default-page: "Load the default page."
       - close: "Close the window."
-  desc: Behavior when the last tab is closed.
+  desc: How to behave when the last tab is closed.
```
Here we prefer 'How...' over a simple noun

mschilli87 commented:
I'm fine with leaving some things 'less consistent'.
I just thought the above changes kind of contradicted each other in a PR attempting to standardize things.
I just didn't know if you prefer one over the other so I asked before submitting the new changes.

I'll finish this off later today or tomorrow.

mschilli87 commented:
I rebased on latest master, split the changes into several commits with related chunks and applied those changes for cases missed so far.
Please give this another thorough round of review before merging.

I kept the previous source of this PR as `standardize_configdata_old` branch in my fork and opened a PR to make it easy for you to see what I changed since the last review: https://github.com/mschilli87/qutebrowser/pull/1.

mschilli87 commented:
I just saw that https://github.com/qutebrowser/qutebrowser/commit/5469a238de3a21635a6aea068c8e61ad8da11def introduced [a placeholder list](https://github.com/qutebrowser/qutebrowser/blob/5469a238de3a21635a6aea068c8e61ad8da11def/qutebrowser/config/configdata.yml#L773-L784) that didn't match the style of [existing ones](https://github.com/qutebrowser/qutebrowser/tree/5469a238de3a21635a6aea068c8e61ad8da11def/qutebrowser/config/configdata.yml#L1278-L1290).

So I rebased again and added https://github.com/qutebrowser/qutebrowser/pull/3181/commits/7e07f5c9965e109a87c6661e8eb61a57020efa2c to this PR.

The-Compiler commented on .github/CODEOWNERS:
```diff
@@ -1,5 +1,6 @@
 qutebrowser/browser/history.py  @rcorre
 qutebrowser/completion/*  @rcorre
+qutebrowser/config/configdata.yml  @mschilli87
```
I'd rather keep this somewhat sorted, but I'll fix this up after merging.

mschilli87 replied:
I thought is was alphabetically sorted. That's how I placed the line.
Just now saw that `tests/` is *after* `qutebrowser/`. :smile:

The-Compiler replied:
Ah, fair enough :wink: I'd rather keep it sorted by topics/people roughly, and did so in 6233358b718b73b1edbe2d80c4c8606c22855e09 @The-Compiler

The-Compiler commented on qutebrowser/config/configdata.yml:
```diff
@@ -1274,20 +1276,20 @@ tabs.title.format:
-    * `{title}`: The title of the current web page
+    * `{perc}`: Percentage as a string like `[10%]`.
+    * `{perc_raw}`: Raw percentage, e.g. `10`
+    * `{title}`: Title of the current web page
```
Those should have a trailing period for consistency, but I'll fix that up after merging.

The-Compiler replied:
Fixed in c22a27d47f8f0f331ff61bc089d20d5b0ffbb416

The-Compiler commented:
Thanks a lot for the persistence and the sharp eye, I definitely like to keep things consistent! 👍 

I was wondering whether it'd be a good idea to have some short "writing guidelines" as a comment at the top of the file - if nothing else, we could point people at those if a contribution introduces a new inconsistency. I thought it shouldn't hold this PR back, but maybe would be nice to have? What do you think?

mschilli87 commented:
I thought the same but I'm not sure when I'll have the time.
Feel free to open an issue and assign me if you want. :wink:

Otherwise I would probably see how 'bad' it is in future PR.
Maybe having a consistent style already helps contributors to figure out the 'right' style by themselves.

The-Compiler commented:
Yeah, I thought so too - chances are people won't read it anyways 😆 

Let's just wait and see how things go then.
